src/benchq/problem_ingestion/molecule_instance_generation.py

```diff
@@ -318,19 +236,79 @@ def _get_molecular_data(self):
 
         return molecular_data
 
-    def _set_frozen_core_orbitals(self, molecular_data) -> mp.mp2.MP2:
+    def _set_frozen_core_orbitals(self, mean_field_object) -> mp.mp2.MP2:
         """
         Set auto-generated chemical core orbitals.
 
         Args:
-            molecular_data: PyscfMolecularData object.
+            mean_field_object: The PySCF meanfield object.
+
         Returns
             The mp2 object.
-
         """
-        mp2 = mp.MP2(molecular_data._pyscf_data["scf"]).set_frozen()
+        mp2 = mp.MP2(mean_field_object).set_frozen()
         return mp2
 
+    def _truncate_with_fno(
+        self,
+        molecule: gto.Mole,
+        mean_field_object: scf.hf.SCF,
+    ) -> Tuple[gto.Mole, scf.hf.SCF]:
+        """Truncates a meanfield object by reducing the virtual space using
+        the frozen natural orbital (FNO) method.
+
+        Args:
+            molecule: The PySCF molecule object.
+            mean_field_object: The meanfield object to be truncated.
+
+        Returns:
+            Tuple whose first element is the PySCF molecule object after FNO vertual space
+              reduction and whose second is the meanfield object containing the SCF solution.
+        """
+
+        if molecule.multiplicity != 1:
+            raise ValueError("RO-MP2 is not available.")
+
+        n_frozen_core_orbitals = 0
+
+        if self.freeze_core and self.occupied_indices:
+            raise ValueError(
+                "Both freeze core and occupied_indices were set!"
+                "Those options are exclusive. Please select either one."
+            )
+
+        elif self.freeze_core and not self.occupied_indices:
+            mp2 = self._set_frozen_core_orbitals(mean_field_object)
+            n_frozen_core_orbitals = mp2.frozen
+
+        elif self.occupied_indices and not self.freeze_core:
+            mp2 = mp.MP2(mean_field_object).set(frozen=self.occupied_indices)
+            n_frozen_core_orbitals = len(list(self.occupied_indices))
+
+        else:
+            mp2 = mp.MP2(mean_field_object)
+
+        mp2.verbose = 4
+        mp2.run()
+
+        frozen_natural_orbitals, natural_orbital_coefficients = mp2.make_fno(
+            self.fno_threshold,
+            self.fno_percentage_occupation_number,
+            self.fno_n_virtual_natural_orbitals,
+        )
+
+        if len(frozen_natural_orbitals) != 0:
+            mean_field_object.mo_coeff = natural_orbital_coefficients[
+                :, : -len(frozen_natural_orbitals)
+            ]
+        else:
+            mean_field_object.mo_coeff = natural_orbital_coefficients
+
+        # Calculate the number of orbitals after truncation with fno
+        molecule._nao = mean_field_object.mo_coeff.shape[1]
```

Comment:
It may not make a lot of difference, but could you please use the "setter" method rather than accessing the private attribute? I.e. `molecule.nao = ...`

Reply:
Of course! This a way I it should have done! Fixed.